lib/util/cache/package/decorator.ts

```diff
@@ -114,7 +114,10 @@ export function cache<T>({
 
     const result = await callback();
 
-    await packageCache.set(finalNamespace, finalKey, result, ttlMinutes);
+    // only cache if we got a valid result
+    if (result !== undefined) {
+      await packageCache.set(finalNamespace, finalKey, result, ttlMinutes);
+    }
```

Comment:
@JamieMagee WDYT?

Reply:
Yeah, this is a good idea. I just have a couple of questions about the tests.

Reply:
I think we should move the cache changes to a separate pr. I think it can have side effects to exiting caching.

So please move this to a sepearate pr and target the v26 branch, just to be safe.

Reply:
Hmm, I'd like to understand what these are :)